src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -165,6 +166,15 @@ public function createAction(Request $request)
 
         $parentId = (int) $request->query->get('id_parent', $this->configuration->getInt('PS_HOME_CATEGORY'));
 
+        /*
+         * Parent category can be root category only if you specifically click to add root category.
+         * In all other cases it should be at least home category(Or one of it's children).
+         */
+        $configRootCategory = $this->configuration->getInt('PS_ROOT_CATEGORY');
```

Comment:
That's weird it means root category should never be selected as parent right? So how comes it can be selected as parent of this route?
Since root category have their own route and controller we shouldn't bother handling them here? Or do we?

It seems like it's fixing a bug which root cause is actually the route generation, but maybe it's better to have a safeguard here so that the controller is more resilient?